pymeasure/adapters/serial.py

```diff
@@ -103,9 +103,12 @@ def _read_bytes(self, count, break_on_termchar, **kwargs):
             return self.connection.read_until(self.read_termination.encode(),
                                               count if count > 0 else None,
                                               **kwargs)
+        elif count >= 0:
+            return self.connection.read(count, **kwargs)
         else:
-            # At -1 we read a very large number of bytes, which can be considered the whole buffer.
-            return self.connection.read(1e99 if count == -1 else count, **kwargs)
+            # For -1 we empty the buffer completely
+            num_bytes = self.connection.in_waiting + 100  # Read what's already queued + some extra
+            return self.connection.read(num_bytes, **kwargs)
```

Comment:
Let us return to that original version, which really reads until a timeout happens:
```suggestion
            return b"\n".join(self.connection.readlines(**kwargs))
```

Your proposed solution fails, if a device sends 101 (or more) bytes as we read only the first 100 bytes.

Reply:
I was testing with return b"\n".join(self.connection.readlines(**kwargs)) and it seems the newline character is not removed from the result, so just "".join(...) would work.

But also double the timeout occurs before the function returns (which is really an issue inside pyserial it seems)

Reply:
Pushed a new version, reading by chucks does seem to solve it everything, it gives us the most control. I also included my looped test, let me know if it should be left out instead.

@bmoneke do you concur?

Reply:
The first suggestion (not sent), was reading chunks, but I thought the newline thing shorter.

In the end, your solution is nice and understandable (not that "read lines" magic).